### PR TITLE
Refactor superdesk to use superdesk-core library

### PR DESCRIPTION
**Pull request by amagdas**
- superdesk forlder moved to https://github.com/superdesk/superdesk-core
- ContentAPI moved to https://github.com/superdesk/superdesk-content-api
- users moved to core
- some ingest providers were moved out from core to apps/io/
- aap_mm_datalayer moved to apps/aap_mm

### CODE REVIEW

File: server/apps/prepopulate/app_initialization_test.py [hunk 1]
```diff
@@ -1,52 +1,46 @@
...
+class AppInitializeWithDataCommandTestCase(SuperdeskTestCase):
 
     def test_app_initialization(self):
-        with self.app.app_context():
-            command = AppInitializeWithDataCommand()
-            result = command.run()
-            self.assertEquals(result, 0)
+        command = AppInitializeWithDataCommand()
+        result = command.run()
+        self.assertEquals(result, 0)
```
Comment (plamut) [inline server/apps/prepopulate/app_initialization_test.py]: No sure if this is the right PR for changing this, `assertEquals()` is deprecated and should be renamed to `assertEqual()`.

File: server/apps/publish/formatters/newsml_1_2_formatter_tests.py [hunk 2]
```diff
@@ -111,33 +110,30 @@ def test_format_news_management_for_corrections(self):
...
+            get('FormalName'), '02011001')
+        self.assertEquals(
+            self.newsml.findall('NewsComponent/NewsComponent/DescriptiveMetadata/SubjectCode/Subject')[1].
+            get('FormalName'), '02011002')
+        self.assertEquals(self.newsml.find('NewsComponent/NewsComponent/DescriptiveMetadata/Property').
+                          get('Value'), 'a')
+        self.assertEquals(
+            self.newsml.findall('NewsComponent/NewsComponent/NewsComponent/ContentItem/DataContent')[0].
+            text, 'sample abstract')
+        self.assertEquals(
```
Comment (plamut) [inline server/apps/publish/formatters/newsml_1_2_formatter_tests.py]: `assertEquals()` is deperecated (if it applies for this PR).

File: server/apps/publish/formatters/newsml_g2_formatter_test.py [hunk 3]
```diff
@@ -356,95 +356,88 @@ def setUp(self):
...
+        self.assertEquals(xml.find(
+            '{http://iptc.org/std/nar/2006-10-01/}itemSet/{http://iptc.org/std/nar/2006-10-01/}newsItem/' +
+            '{http://iptc.org/std/nar/2006-10-01/}itemMeta/{http://iptc.org/std/nar/2006-10-01/}provider/' +
+            '{http://iptc.org/std/nar/2006-10-01/}name').text,
+            'sourcefabric.org')
+        self.assertEquals(xml.find(
+            '{http://iptc.org/std/nar/2006-10-01/}itemSet/{http://iptc.org/std/nar/2006-10-01/}newsItem/' +
+            '{http://iptc.org/std/nar/2006-10-01/}contentMeta/{http://iptc.org/std/nar/2006-10-01/}headline').text,
+            'This is a test headline')
+        self.assertEquals(xml.find(
```
Comment (plamut) [inline server/apps/publish/formatters/newsml_g2_formatter_test.py]: `assertEquals()` deprecation notice.

(will stop mentioning these in order to not shadow the essence of the changes)

**Comment by plamut** [comment]
This one is so massive that it really slows down the browser initially. :)
But from what I can see, the last 2/3rds of it is really just deleted files that have been moved elsewhere? And by refactoring the application object, a lot of `with self.app.app_context()` lines in the tests were able to get removed?

I think this is great stuff that will make the project and its dependencies/add-ons more manageable. :heart: 

When this is finished, I suggest adding the instructions for developers on how to update their local environments, repository paths etc.

Also, what is the time line for this being merged? Should we, at some point, briefly stop writing new code in order to make the transition easier, and only continue once the new repositories are in place and the system working? Or will the transition be as easy as updating the remotes in git config?

Thanks!

File: server/apps/prepopulate/app_initialization_test.py (hunk 1 above)
Comment (amagdas) [inline server/apps/prepopulate/app_initialization_test.py]: This is to be tackled in a different PR, this PR is huge already as it is.

**Comment by petrjasek** [comment]
:+1:

File: server/apps/macros/macro_register.py [hunk 4]
```diff
@@ -41,8 +41,10 @@ def load_module(module):
 def register_macros():
     macro_modules = [sys.modules[m] for m in sys.modules.keys()
                      if 'macros.' in m and
+                     'callback' in dir(sys.modules[m]) and
                      not m.endswith('_test')
                      and not m.startswith('__')]
+    print(macro_modules)
```
Comment (mugurrus) [inline server/apps/macros/macro_register.py]: Remove print

**Comment by mugurrus** [comment]
:+1:

**Comment by plamut** [comment]
OK, so how to migrate to the new structure? Any specific steps that need to be taken or... ?

**Comment by amagdas** [comment]
Check your email pls.

**Comment by plamut** [comment]
Sorry, didn't see that, closed it when I finished my work day. Will read through it, thanks.
